src/pieces/commands/onboarding.py

```diff
@@ -0,0 +1,191 @@
+
+class OnboardingStep(BasedOnboardingStep):
+    def __init__(self,text:str, validation: Callable) -> None:
+        self.text = text
+        self.validation = validation
+    def run(self, console:Console):
+        console.print(Markdown(self.text))
+        self.click_to_continue(console)
+        
+        while True:
+            validation, message_failed = self.validation()
+            if validation:
+                break  # Exit the loop if validation is successful
+            else:
+                console.print(message_failed)
+                self.click_to_continue(console)
+        
+        return True
+
+class OnboardingCommandStep(BasedOnboardingStep):
+    def __init__(self,text, predicted_text: str) -> None:
+        self.text = text
+        self.predicted_text = predicted_text
+    def run(self, console:Console):
+        console.print(Markdown(self.text))
+        # self.click_to_continue(console)
+        
+        user_input = input(get_prompt()).strip()
+        while user_input != self.predicted_text:
+            if user_input == "exit":
+                sys.exit(1)
+            console.print(Markdown(f"❌ Wrong command entered, You should type: `{self.predicted_text}`"))
+            user_input = input(get_prompt()).strip()
+
+        run_command(*extract_text(user_input.removeprefix("pieces ")))
+
+        return True
+
+def create_snippet_one_validation():
+    text = pyperclip.paste()
+    normalized_s1 = '\n'.join(line.strip() for line in text.strip().splitlines())
+    normalized_s2 = '\n'.join(line.strip() for line in demo_snippet.strip().splitlines())
+
+    return normalized_s1 == normalized_s2 , "Seems you did not copy the snippet yet"
+
+def onboarding_command(**kwargs):
+    console = Console()
+    step_number = 1
+    steps = {
+        "Step 1: Creating a snippet":[
+            OnboardingStep("Let's get started by saving a snippet\n"
+                    "copy the following snippet python snippet: \n"
+                    f"```python\n{demo_snippet}\n```",
+                    create_snippet_one_validation
+                ),
+            OnboardingCommandStep(
+                "You can save a snippet by typing `pieces create`",
+                "pieces create"
+            )
+        ],
+        "Step 2: Opening the created snippet":[
+            OnboardingCommandStep(
+                "Now let's checkout all the saved snippets by typing `pieces list`",
+                "pieces list"
+            )
+        ],
+        "Step 3: Run in loop":[
+            OnboardingStep(
+                "Running in loop help you to create a session with the Copilot which we will be convering " 
+                "or perform an opration on the same snippet multiple times which will increase your **productivity**",
+                lambda: (True, "")
+            ),
+            OnboardingCommandStep(
+                "You can run in loop by typing `pieces run` Don't forget to exit the loop by typing `exit`",
```

Comment:
You can run in loop by typing `pieces run`. Don't forget to exit the loop by typing `exit`